src/core/server/http/router/route.ts

```diff
@@ -18,7 +18,23 @@
  */
 
 import { ObjectType, Schema } from '@kbn/config-schema';
-export type RouteMethod = 'GET' | 'POST' | 'PUT' | 'DELETE';
+export type RouteMethod = 'get' | 'post' | 'put' | 'delete';
+
+export interface RouteConfigOptions {
+  /**
+   * A flag shows that authentication for a route:
+   * enabled  when true
+   * disabled when false
+   *
+   * Enabled by default.
+   */
+  authRequired?: boolean;
+
+  /**
+   * Additional meta-data information to attach fot the route.
+   */
+  tags?: string[];
```

Comment:
I think using Set could be error-prone because strings are iterable in js
```js
new Set('my:tag')    // wrong declaration, don't throw
new Set(['my:tag']) // correct declaration, but not intuitive
```

Reply:
Yeah, it's arguable and may not be the best choice in some cases, don't have a strong opinion on that, feel free to pick whatever you think is better.

P.S. We decided to move with `Set` for a similar thing in [`EcnrytpedSavedObjects`](https://github.com/elastic/kibana/blob/master/x-pack/test/plugin_api_integration/plugins/encrypted_saved_objects/index.ts#L44) to signify uniqueness of the setting. It's not super important, just slowly transforming my coding habits towards `string[] ---> Set<string>` and `Record<string, string> ---> Map<string, string>` where appropriate :slightly_smiling_face: